### DISCUSSION TITLE
(noob) Issues with config/OpenVPN with TorGuard

### DISCUSSION THREAD

**Discussion by Hazeel88** [haugene/docker-transmission-openvpn#2276] [Q&A] [answered]
Before reading this, please keep in mind that my skills and experience with programming and VPN settings are at a beginners level. If you are so kindly as to reply with help, I can only hope you will keep this in mind.

Gentlemen, I am in need of your help. 
Inspired by this article [https://www.lejimmy.com/setting-up-transmission-bittorrent-with-openvpn-on-a-synology-nas-and-docker/](url) I am now attempting the same using my TorGuard VPN and a Synology DS920+.
Setting up docker and haugene image was a breeze, but the settings for the actual OpenVPN is giving me a headache. 

These are the settings exported from the container:
[OpenVPN setting.txt](https://github.com/haugene/docker-transmission-openvpn/files/8883585/OpenVPN.setting.txt)
NOTE that some of these settings required a input, so I consulted this link to fill them out: [https://haugene.github.io/docker-transmission-openvpn/config-options/](url)

And this is the error it produces:
[Log.txt](https://github.com/haugene/docker-transmission-openvpn/files/8883586/Log.txt)

My technical knowhow is not enough to solve this, so I humbly ask you gentlemen for assistance.

If more information is needed, please also describe how I can get the information for you.

**Comment by Ye-Yint-Nyo-Hmine** [comment]
I would like to answer but the link for [https://www.lejimmy.com/setting-up-transmission-bittorrent-with-openvpn-on-a-synology-nas-and-docker/](https://github.com/haugene/docker-transmission-openvpn/discussions/url) is not working

**Reply by Hazeel88** [reply]
Here you go:
https://www.lejimmy.com/setting-up-transmission-bittorrent-with-openvpn-on-a-synology-nas-and-docker/
Thought I was being slick with the url code, but must have messed it up.

**Comment by dli7319** [comment]
`curl: (6) Could not resolve host: github.com` suggests you don't have an internet connection or your DNS resolution is not working within the container.
I suggest setting `OPENVPN_CONFIG=dummy` to see if it can download and list the configs.

**Reply by dli7319** [reply]
What are you mounting to `/etc/resolv.conf`?

**Reply by Hazeel88** [reply]
The resolv.conf contains:
`nameserver 8.8.8.8`
`nameserver 8.8.4.4`

This is what I understand "mount" to be.

**Reply by Hazeel88** [reply]
> I suggest setting OPENVPN_CONFIG=dummy to see if it can download and list the configs.

Had to google this, and I think you mean to set `OPENVPN_USERNAME=dummy OPENVPN_password=dummy`

Config: 
[Config export.txt](https://github.com/haugene/docker-transmission-openvpn/files/8886753/Config.export.txt)
Error log:
[Log.txt](https://github.com/haugene/docker-transmission-openvpn/files/8886756/Log.txt)

**Reply by dli7319** [reply]
`OPENVPN_CONFIG` lets you select which servers to use https://haugene.github.io/docker-transmission-openvpn/run-container/#2_it_is_not_mandatory_but_setting_openvpn_config_is_good.
What happens if you go back to your original config but do not mount `/etc/resolv.conf`?

**Reply by dli7319** [reply]
looks to be the same issue as https://github.com/haugene/docker-transmission-openvpn/discussions/1993

**Comment by Hazeel88** [comment, marked as answer]
Reading these responses has made me realize that I need to read the documentation again, do some research on what Docker actually is and its functions before I try again. 

Im going to close this ticket and try to get it working again with more knowledge. If success; post solution here. If fail; ask for help again but with much more information about the project, what I have tried and what exactly I need help with.
